File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cpp
class Solution {
public:
    double findMedianSortedArrays(vector<int>& nums1, vector<int>& nums2) {
        vector<int>nums;
        for(int i=0; i<nums1.size(); i++)
        {
            nums.push_back(nums1[i]);
        }
        
        for(int i=0; i<nums2.size(); i++)
        {
            nums.push_back(nums2[i]);
        }
        
        sort(nums.begin(), nums.end());
        
        int n = nums.size();
        double median;
        if(n%2 == 0)
        {
            int m = n/2;
            int n = m-1;
            median = (nums[m]+nums[n])/2.0;
        }
        else
        {
            int m = n/2;
            median = nums[m];
        }
        
        return median;
    }
};

File: 0004-median-of-two-sorted-arrays/0004-median-of-two-sorted-arrays.cs
public class Solution {
    public double FindMedianSortedArrays(int[] nums1, int[] nums2) {
        List<int> nums = new List<int>();
        
        // Add elements from nums1 to the list
        foreach (int num in nums1)
        {
            nums.Add(num);
        }
        
        // Add elements from nums2 to the list
        foreach (int num in nums2)
        {
            nums.Add(num);
        }
        
        // Sort the combined list
        nums.Sort();
        
        int n = nums.Count;
        double median;
        
        // If the number of elements is even, take the average of the middle two
        if (n % 2 == 0)
        {
            int mid1 = n / 2;
            int mid2 = mid1 - 1;
            median = (nums[mid1] + nums[mid2]) / 2.0;
        }
        else
        {
            // If the number of elements is odd, take the middle element
            int mid = n / 2;
            median = nums[mid];
        }
        
        return median;
    }
}